src/storage/v2/durability/durability.cpp

```diff
@@ -306,6 +308,28 @@ void RecoverUniqueConstraints(const RecoveredIndicesAndConstraints::ConstraintsM
   spdlog::info("Constraints are recreated from metadata.");
 }
 
+void RecoverTypeConstraints(const RecoveredIndicesAndConstraints::ConstraintsMetadata &constraints_metadata,
+                            Constraints *constraints, utils::SkipList<Vertex> *vertices, NameIdMapper *name_id_mapper,
+                            const std::optional<ParallelizedSchemaCreationInfo> & /**/) {
+  // TODO: parallel recovery
+  spdlog::info("Recreating {} type constraints from metadata.", constraints_metadata.type.size());
+  for (const auto &[label, property, type] : constraints_metadata.type) {
+    if (constraints->type_constraints_->ConstraintExists(label, property)) {
+      throw RecoveryFailure("The type constraint already exists!");
+    }
+
+    if (auto violation = TypeConstraints::ValidateVerticesOnConstraint(vertices->access(), label, property, type);
+        violation.has_value()) {
+      throw RecoveryFailure("Type constraint recovery failed because it couldn't be validated!");
+    }
```

Comment:
I would like to do the validation for all `constraints_metadata.type` in one scan rather than N scans.